adaptdl/adaptdl/torch/scaling_rules_test.py

```diff
@@ -37,11 +38,66 @@ def test_object():
     obj.set_accum_scale(3.0)
     assert(obj.accum_scale == 3.0)
     assert(np.isclose(obj.gain(2.0), 1.0))
-    obj._state['var_avg'] = 3.0
-    obj._state['norm_avg'] = 1.0
+    obj.state['var_avg'] = 3.0
+    obj.state['norm_avg'] = 1.0
     assert(np.isclose(obj.gain(3.0), 2.0))
 
 
+def test_scaling_rules_1():
+    """test AdaScale lr factors"""
+    adp = Mock(require_backward_grad_sync=True)
+    gns = Mock(state={"var_avg": np.asarray([1, 0, 0, 2]),
+                      "sqr_avg": np.asarray([-1, 0, -1, 1])})
+    adp.gns = gns
+    adascale = AdaScale()
+    adascale.adp = adp
+    np.testing.assert_equal(adascale.calculate_lr_factors(2),
+                            [2., 2., 2., 1.5])
+
+
+def test_scaling_rules_2():
+    """test LinearScale lr factors"""
+    adp = Mock(require_backward_grad_sync=True)
+    opm = Mock(param_groups=[1, 0, 2, -1])
+    gns = Mock(optimizer=opm)
+    adp.gns = gns
+    linearscale = LinearScale()
+    linearscale.adp = adp
+    np.testing.assert_equal(linearscale.calculate_lr_factors(1),
+                            [1., 1., 1., 1.])
+
+
+def test_scaling_rules_3():
+    """test SqrtScale lr factors"""
+    adp = Mock(require_backward_grad_sync=True)
+    opm = Mock(param_groups=[1, 0, 2, -1])
+    gns = Mock(optimizer=opm)
+    adp.gns = gns
+    sqrtscale = SqrtScale()
+    sqrtscale.adp = adp
+    np.testing.assert_equal(sqrtscale.calculate_lr_factors(9),
```

Comment:
Nit: should probably test these on more than one value. A for loop over a range should be fine I think.